### ISSUE TITLE
Rule proposal: `no-plain-function-expression`

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#1716]
### Description

A recurring bug that I sometimes introduce involves forgetting the `()` in a function call. The resulting JS is valid, but the code only test for _existance_ of said function, which is worthless.

I guess this rule would have a risk of false positives, so the functions that are to be checked should probably be reduced ones defined in the user's own code to avoid triggering on things like polyfills which have a legimitate use case for testing for the existance of an function.

### Fail

```js
const isTrue = () => true;
console.log(isTrue ? 'true' : 'false');
```

### Pass

```js
const isTrue = () => true;
console.log(isTrue() ? 'true' : 'false');
```

**Comment by silverwind** [comment]
Those TypeScript rules look to be very close to what I want, but I wonder if it possible to have such a rule for JavaScript too.

Rule would definitely be more generic to detect on all cases of constant values used in a boolean context, at least in simple cases like `const` assignments.

**Comment by silverwind** [comment]
Closing as I think the proposal is too limited in scope and because it will definitely require type information which I guess is not easily implementable for JS.

**Comment by fisker** [comment]
> which I guess is not easily implementable for JS.

Doable for local functions.

**Comment by silverwind** [comment]
> Doable for local functions.

Interesting. How about functions imported from other files via named import? Is such information in the AST or do you only have the current file AST?

**Comment by fisker** [comment]
only see the current file's AST?
